### ISSUE TITLE
Client fails to upload synchronisation repo but server thinks it has

### ISSUE THREAD

**Issue by akumria** [haiwen/seafile#1277]
Hi,

Seafile Client 4.1.6 and 4.2.3
Seafile Server 4.0.5

I have 4 computers, each running a variety of the the above client versions. On only one computer do I have a problem. Initially running 4.1.6 and then in an effort to determine if the bug has been fixed was upgraded to 4.2.3 with no change.

Log lines below:

[06/10/15 00:20:04] sync-mgr.c(638): Repo 'A' sync state transition from 'synchronized' to 'uploading'.
[06/10/15 00:20:04] http-tx-mgr.c(2646): Upload with HTTP sync protocol version 1.
[06/10/15 00:20:04] http-tx-mgr.c(926): Transfer repo '63fd16cb': ('normal', 'init') --> ('normal', 'check')
[06/10/15 00:20:09] http-tx-mgr.c(926): Transfer repo '63fd16cb': ('normal', 'check') --> ('normal', 'commit')
[06/10/15 00:20:10] http-tx-mgr.c(926): Transfer repo '63fd16cb': ('normal', 'commit') --> ('normal', 'fs')
[06/10/15 00:20:12] http-tx-mgr.c(865): libcurl failed to POST http://files.A.com/seafhttp/repo/63fd16cb-46fa-4775-bf7d-539969ae1510/check-fs/: Send failed since rewinding of the data stream failed.
[06/10/15 00:20:12] http-tx-mgr.c(2707): Failed to check fs list for repo 63fd16cb.
[06/10/15 00:20:12] http-tx-mgr.c(926): Transfer repo '63fd16cb': ('normal', 'fs') --> ('error', 'finished')
[06/10/15 00:20:12] sync-mgr.c(705): Repo 'A' sync state transition from uploading to 'error': 'Error occured in upload.'.

On the server, I see:

27.33.193.158 - - [09/Jun/2015:23:22:51 +0000] "GET /seafhttp/repo/e09fe5d9-3923-4fd5-a8c7-42033b350e9a/commit/HEAD HTTP/1.1" 200 406 "-" "Seafile/4.2.3 (Windows NT)"
27.33.193.158 - - [09/Jun/2015:23:22:54 +0000] "GET /seafhttp/repo/982981e7-da69-4632-b884-ea56a68d3118/commit/HEAD HTTP/1.1" 200 480 "-" "Seafile/4.2.3 (Linux)"
27.33.193.158 - - [09/Jun/2015:23:22:57 +0000] "GET /seafhttp/repo/63fd16cb-46fa-4775-bf7d-539969ae1510/commit/HEAD HTTP/1.1" 200 480 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:01 +0000] "GET /ajax/unseen-notices-count/?_=1433892184456 HTTP/1.1" 401 714 "https://files.A.com/" "Mozilla/5.0 (X11; Linux x86_64; rv:37.0) Gecko/20100101 Firefox/37.0"
87.115.214.73 - - [09/Jun/2015:23:23:03 +0000] "GET /seafhttp/repo/91a49db5-b883-43bc-bb0a-60cb907235ee/commit/HEAD HTTP/1.1" 200 406 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:04 +0000] "GET /seafhttp/repo/8d00599a-79c3-4c8b-945a-d18030f9449a/commit/HEAD HTTP/1.1" 200 264 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:07 +0000] "GET /seafhttp/repo/a19cabec-8e9e-48f6-94f0-56cc91cb64aa/commit/HEAD HTTP/1.1" 200 264 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:09 +0000] "GET /seafhttp/repo/63fd16cb-46fa-4775-bf7d-539969ae1510/permission-check/?op=upload&client_id=9ad38da9b2369cf1c89702a1360ddbe55f879f2f&client_name=anand%40bhaga HTTP/1.1" 200 246 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:09 +0000] "GET /seafhttp/repo/63fd16cb-46fa-4775-bf7d-539969ae1510/quota-check/?delta=1793 HTTP/1.1" 200 104 "-" "Seafile/4.2.3 (Linux)"
87.115.214.73 - - [09/Jun/2015:23:23:10 +0000] "PUT /seafhttp/repo/63fd16cb-46fa-4775-bf7d-539969ae1510/commit/6080fa79d28ee9a6c60fc68ff467a2c79d803b9c HTTP/1.1" 200 212 "-" "Seafile/4.2.3 (Linux)"

Any debugging assistance would be appreciated.

Thanks,
Anand

**Comment by killing** [comment]
This issue should be related to
http://curl.haxx.se/libcurl/c/CURLOPT_SEEKFUNCTION.html
Can you find out why your data transfer is interrupted?

2015-06-10 7:24 GMT+08:00 Anand Kumria <notifications@github.com>:

**Comment by akumria** [comment]
I'm happy to, how might I go about doing that?

**Comment by klemens** [comment]
Today I also got the "Send failed since rewinding of the data stream failed." error on one of my machines. It turned out the reason was that the client was using http while the server would "force" https by 301-redirecting all http-requests to https, which seems to confuse curl. As soon as I changed the server-address to https, everything worked fine.

I am not sure if this is related to your problem however.
